pkg/controller/shared_state.go

```diff
@@ -1565,3 +1573,62 @@ func (c *SharedState) getNodesTopologySelector(ctx context.Context, nodes []v1.N
 	}
 	return topologySelector
 }
+
+func (c *SharedState) addNodeToAvailableAttachmentsMap(ctx context.Context, nodeName string, nodeLables map[string]string) {
+	if _, ok := c.availableAttachmentsMap.Load(nodeName); !ok {
+		capacity, err := azureutils.GetNodeRemainingDiskCount(ctx, c.cachedClient, nodeName)
+		if err != nil {
+			klog.Errorf("Failed to get node(%s) remaining disk count with error: %v", nodeName, err)
+			// store the maximum capacity if an entry for the node doesn't exist.
+			capacity, err = azureutils.GetNodeMaxDiskCountWithLabels(nodeLables)
+			if err != nil {
+				klog.Errorf("Failed to add node(%s) in availableAttachmentsMap, because get capacity of available attachments is failed with error: %v", nodeName, err)
+				return
+			}
+		}
+		var count atomic.Int32
+		count.Store(int32(capacity))
+		klog.Infof("Added node(%s) to availableAttachmentsMap with capacity: %d", nodeName, capacity)
+		c.availableAttachmentsMap.LoadOrStore(nodeName, &count)
+	}
+
+	klog.Infof("Node(%s) is already in availableAttachmentsMap. Don't need to add it again", nodeName)
```

Comment:
This log isn't all that useful for debugging and can be removed.

Reply:
Done.